### DISCUSSION TITLE
How to take screenshot with DeckGL with Google Maps as the base layer?

### DISCUSSION THREAD

**Discussion by vivek091195** [visgl/deck.gl#6264] [Q&A]
In my application, I am using Deck.gl, and the base layer is Google Maps. I know that Deck.gl maintains a reference to its canvas. How can I make use of it to take a screenshot of plotted layers on the map?
Here is my [code](https://jsfiddle.net/vivekkumar091195/ty1mj6ud/5/)
On clicking the screenshot button, it's downloading a blank image with no content. What am I missing here? Thanks in advance for your help!

**Comment by Pessimistress** [comment]
https://github.com/visgl/deck.gl/issues/4436

**Comment by chrisgervang** [comment]
Adding `deckOverlay._deck.redraw(true);` to the first line of your screenshot function helps capture a screenshot of the deck overlay, however the google map is rendered under this somehow.. and I couldn't find a canvas for their map. I'm not sure if it's possible to create a screenshot of the whole map unless they render to a canvas, and then you'd still need to combine deck and google's canvas to get the complete picture.

This is what I captured:
![Download (4)](https://user-images.githubusercontent.com/2461547/136280369-a97c21b9-f122-4f58-a946-28263c20a20b.jpg)
